File: 6 - Sistemas Distribuidos/Exemplos/calculadora_rmi/java/Calculadora.java
/**
 * Define a interface para uma calculadora remota
 * Autor: Rodrigo Campiolo
 * Data: 22/11/2006
 */

import java.rmi.Remote;
import java.rmi.RemoteException;

public interface Calculadora extends Remote {
    public int soma (int a, int b) throws RemoteException;
    public int subtrai (int a, int b) throws RemoteException;
    public int divide (int a, int b) throws RemoteException;
    public int multiplica (int a, int b) throws RemoteException;
} //Calculadora

File: 6- Sistemas Distribuidos/Exemplos/calculadora_rmi/python/Calculadora.py
import Pyro4

@Pyro4.expose
class Calculadora:
    def soma(self, a, b):
        return a + b
    
    def subtrai(self, a, b):
        return a - b
    
    def multiplica (self, a, b):
        return a * b
    
    def divide (self, a, b):
        return a // b
